### PR TITLE
travelperk payment profile settings form and save settings

### PR DESCRIPTION
**Pull request by DhaaraniCIT**
## Summary by CodeRabbit

- **New Features**
	- Introduced Travelperk-specific settings and update events to enhance integration capabilities.
	- Added new functionality for Travelperk payment profile settings, including user role management.
- **Enhancements**
	- Improved the onboarding flow for Travelperk integration by refining navigation logic.
	- Enhanced API service configuration for better consistency and reliability.
- **Refactor**
	- Renamed `workspace` properties to `org` in Travelperk configuration models for clearer alignment with domain terminology.
- **Bug Fixes**
	- Adjusted import settings to correctly filter and map expense fields based on user preferences.

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
@coderabbitai review

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts [hunk 1]
```diff
@@ -1,23 +1,91 @@
...
 export interface TravelperkPaymentProfileSettingPost extends TravelperkPaymentProfileSetting { }
 
 export interface TravelperkPaymentProfileSettingGetPaginator extends Paginator {
     results: TravelperkPaymentProfileSettingGet[]
+}
+
+export type TravelPerkPaymetProfileSettingFormOption = {
+    label: string,
+    value: TravelPerkUserRole
+}
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts:29]: any field that uses label and value let's use `SelectFormOption`

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts [hunk 2]
```diff
@@ -1,23 +1,91 @@
...
+
+export class TravelperkPaymentProfileSettingModel {
+
+    static getUserRoles(): TravelPerkPaymetProfileSettingFormOption[] {
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts:33]: getUserRoleOptions

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts [hunk 3]
```diff
@@ -1,23 +1,91 @@
...
+            }
+        ];
+    }
+
+    static createFormGroup(data: TravelperkPaymentProfileSettingGet): FormGroup {
+        return new FormGroup ({
+            payment_profile_name: new FormControl(data.payment_profile_name || ''),
+            payment_profile_id: new FormControl(data.payment_profile_id || ''),
+            user_role: new FormControl(data.user_role || null),
+            import_to_fyle: new FormControl(data.import_to_fyle || false)
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts:55]: use camelcase

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts [hunk 4]
```diff
@@ -1,23 +1,91 @@
...
+        });
+    }
+
+    static constructFormArray(arrayData: TravelperkPaymentProfileSettingGet[]): FormGroup[] {
+        const arr:FormGroup[] = [];
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts:60]: better name

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts [hunk 5]
```diff
@@ -1,23 +1,91 @@
...
+    static createFormGroup(data: TravelperkPaymentProfileSettingGet): FormGroup {
+        return new FormGroup ({
+            payment_profile_name: new FormControl(data.payment_profile_name || ''),
+            payment_profile_id: new FormControl(data.payment_profile_id || ''),
+            user_role: new FormControl(data.user_role || null),
+            import_to_fyle: new FormControl(data.import_to_fyle || false)
+        });
+    }
+
+    static constructFormArray(arrayData: TravelperkPaymentProfileSettingGet[]): FormGroup[] {
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts:59]: better name

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts [hunk 6]
```diff
@@ -1,23 +1,91 @@
...
+
+    static mapAPIResponseToFormGroup(travelperkPaymentProfileSettingResponse:TravelperkPaymentProfileSettingGet[] | null): FormGroup {
+        const expenseFieldsArray = travelperkPaymentProfileSettingResponse ? this.constructFormArray(travelperkPaymentProfileSettingResponse) : [] ;
+        return new FormGroup({
+            expenseFields: new FormArray(expenseFieldsArray)
+        });
+    }
+
+    static createPaymentProfileSettingPayload(travelperkPaymentProfileSettingForm: FormGroup){
+        return this.constructPaymentProfileMapping(travelperkPaymentProfileSettingForm.value.expenseFields);
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts:88]: why are we calling 1 more func, can't we add every lines here?

File: src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts [hunk 7]
```diff
@@ -1,23 +1,91 @@
...
+          };
+        });
+
+        return paymentProfileSettings;
+      }
+
+    static mapAPIResponseToFormGroup(travelperkPaymentProfileSettingResponse:TravelperkPaymentProfileSettingGet[] | null): FormGroup {
+        const expenseFieldsArray = travelperkPaymentProfileSettingResponse ? this.constructFormArray(travelperkPaymentProfileSettingResponse) : [] ;
+        return new FormGroup({
+            expenseFields: new FormArray(expenseFieldsArray)
```
Comment (ashwin1111) [inline src/app/core/models/travelperk/travelperk-configuration/travelperk-payment-profile-settings.model.ts:83]: rename according to previous PR comment

File: src/app/integrations/travelperk/travelperk-shared/travelperk-payment-profile-settings/travelperk-payment-profile-settings.component.ts [hunk 8]
```diff
@@ -7,9 +17,84 @@ import { Component, OnInit } from '@angular/core';
...
+
+
+    }, () => {
+      this.isSaveInProgress = false;
+      this.toastService.displayToastMessage(ToastSeverity.ERROR, 'Error saving payment profile settings, please try again later');
+    });
+  }
+
+  save(): void {
+    if (this.paymentProfileMappingForm.valid) {
```
Comment (ashwin1111) [inline src/app/integrations/travelperk/travelperk-shared/travelperk-payment-profile-settings/travelperk-payment-profile-settings.component.ts:79]: btn would be enabled only if it's valid, can add all save logic here

File: src/app/integrations/travelperk/travelperk-shared/travelperk-payment-profile-settings/travelperk-payment-profile-settings.component.ts [hunk 9]
```diff
@@ -7,9 +17,84 @@ import { Component, OnInit } from '@angular/core';
...
+          {
+            phase: this.helper.getPhase(this.isOnboarding),
+            oldState: this.paymentProfileSettings,
+            newState: travelperkPaymentProfileMappingResponse
+          }
+        );
+      }
+
+      if (this.isOnboarding) {
+        this.workspaceService.setOnboardingState(Sage300OnboardingState.ADVANCED_SETTINGS);
```
Comment (ashwin1111) [inline src/app/integrations/travelperk/travelperk-shared/travelperk-payment-profile-settings/travelperk-payment-profile-settings.component.ts:67]: travel perk onboarding state

Comment (DhaaraniCIT) [reply src/app/integrations/travelperk/travelperk-shared/travelperk-payment-profile-settings/travelperk-payment-profile-settings.component.ts:67]: fixed in next PR

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**